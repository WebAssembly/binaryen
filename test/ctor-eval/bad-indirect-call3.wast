(module
  (type $funcref_=>_none (func (param funcref)))
  (memory 256 256)
  (data (i32.const 10) "waka waka waka waka waka")
  (table funcref (elem $callee))
  (export "sig_mismatch" (func $sig_mismatch))
  (func $callee (param $0 externref)
    (i32.store8 (i32.const 40) (i32.const 67))
  )
  (func $sig_mismatch
    (call_indirect (type $funcref_=>_none) ;; unsafe to call, signature mismatch
      (ref.null func)
      (i32.const 0)
    )
  )
)
