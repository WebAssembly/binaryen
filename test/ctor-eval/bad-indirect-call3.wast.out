(module
 (type $none_=>_none (func))
 (type $funcref_=>_none (func (param funcref)))
 (type $externref_=>_none (func (param externref)))
 (memory $0 256 256)
 (data (i32.const 10) "waka waka waka waka waka")
 (table $0 1 1 funcref)
 (elem (i32.const 0) $callee)
 (export "sig_mismatch" (func $sig_mismatch))
 (func $callee (param $0 externref)
  (i32.store8
   (i32.const 40)
   (i32.const 67)
  )
 )
 (func $sig_mismatch
  (call_indirect $0 (type $funcref_=>_none)
   (ref.null func)
   (i32.const 0)
  )
 )
)
